File: Excel/486 Challenge.py
import pandas as pd

path = "486 Create Integer Intervals.xlsx"
input = pd.read_excel(path, usecols="A")
test = pd.read_excel(path, usecols="B")

def group_consecutive(number_string):
    numbers = [int(num) for num in number_string.split(",")]
    numbers.sort()
    ranges = []
    start = end = numbers[0]
    
    for i in range(1, len(numbers)):
        if numbers[i] - numbers[i-1] == 1:
            end = numbers[i]
        else:
            if start == end:
                ranges.append(str(start))
            else:
                ranges.append(f"{start}-{end}")
            start = end = numbers[i]
    
    if start == end:
        ranges.append(str(start))
    else:
        ranges.append(f"{start}-{end}")
    return ", ".join(ranges)

result = input.copy()
result["Answer Expected"] = result["Problem"].map(group_consecutive)

print(result["Answer Expected"].equals(test["Answer Expected"])) # True

File: Excel/400-499/486/486 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/486 Create Integer Intervals.xlsx"
input = read_excel(path, range = "A1:A8")
test  = read_excel(path, range = "B1:B8")

group_consecutive = function(number_string) {
  numbers <- str_split(number_string, ",") %>%
    unlist() %>%
    as.numeric()
  
  tibble(numbers = sort(numbers)) %>%
    mutate(group = cumsum(c(TRUE, diff(numbers) != 1))) %>%
    summarise(range = if_else(n() > 1, 
                              paste0(min(numbers), "-", max(numbers)), 
                              as.character(numbers[1])), .by = group) %>%
    pull(range) %>%
    paste(collapse = ", ")
}

result = input %>%
  mutate(`Answer Expected` = map_chr(Problem, group_consecutive))

identical(result$`Answer Expected`, test$`Answer Expected`)
#> [1] TRUE